packages/cfd/src/features/types/components.types.ts

```diff
@@ -0,0 +1,342 @@
+import { TCFDPlatform, TTokens, TTradingPlatformAvailableAccount, TAccountCategory, TMarketType } from './shared.types';
+
+type TOnSubmit = (
+    index: number,
+    value: TFormValues,
+    setSubmitting: (isSubmitting: boolean) => void,
+    is_dirty?: boolean
+) => void;
+
+// cfd-account-copy
+export type TCFDAccountCopy = {
+    text?: string;
+    className: string;
+};
+
+// cfd-personal-details-form
+export type TCFDPersonalDetailsFormProps = {
+    index: number;
+    form_error?: string;
+    is_loading: boolean;
+    onSubmit: TOnSubmit;
+    initial_values: TFormValues;
+    changeable_fields?: string[];
+    residence_list: ResidenceList;
+};
+
+export type TValidatePersonalDetailsParams = {
+    values: TFormValues;
+    residence_list: ResidenceList;
+    account_opening_reason: TAccountOpeningReasonList;
+};
+
+export type TFindDefaultValuesInResidenceList = (params: {
+    citizen_text: GetSettings['citizen'];
+    tax_residence_text: GetSettings['tax_residence'];
+    place_of_birth_text?: GetSettings['place_of_birth'];
+    residence_list: ResidenceList;
+}) => {
+    citizen?: GetSettings['citizen'];
+    tax_residence?: GetSettings['tax_residence'];
+    place_of_birth?: GetSettings['place_of_birth'];
+};
+
+export type TCFDInputFieldProps = React.ComponentProps<'input'> & {
+    optional?: boolean;
+};
+
+export type TFormValues = { [key: string]: string };
+
+// cfd-poa
+export type TCFDPOA = {
+    index: number;
+    onSave: (index: number, values: FormikValues) => void;
+    onSubmit: (index: number, values: FormikValues) => void;
+};
+
+// cfd-poi
+type TPOIState = {
+    poi_state?: GetAccountStatus['authentication'];
+};
+
+export type TCFDPOIProps = {
+    index: number;
+    height: string;
+    value: TPOIState;
+    onSave: (index: number, values: TPOIState) => void;
+    onSubmit: (index: number, value: TPOIState) => void;
+    removeNotificationByKey: TCoreStores['notifications']['removeNotificationByKey'];
+    jurisdiction_selected_shortcode: DetailsOfEachMT5Loginid['landing_company_short'];
+    removeNotificationMessage: TCoreStores['notifications']['removeNotificationMessage'];
+    addNotificationMessageByKey: TCoreStores['notifications']['addNotificationMessageByKey'];
+};
+
+// specbox
+export type TSpecBoxProps = {
+    value?: string;
+    is_bold?: boolean;
+    is_broker?: boolean;
+};
+
+// success-dialog
+export type TCheckmark = {
+    className: string;
+};
+
+export type TSuccessDialog = {
+    width: React.CSSProperties['width'];
+    title?: string;
+    is_open: boolean;
+    icon_size: string;
+    icon: ReactElement;
+    has_cancel: boolean;
+    has_submit: boolean;
+    text_submit?: string;
+    text_cancel?: string;
+    onCancel?: () => void;
+    onSubmit?: () => void;
+    has_close_icon: boolean;
+    toggleModal: () => void;
+    classNameMessage?: string;
+    is_medium_button?: boolean;
+    message: ReactElement | string;
+    heading?: ReactElement | string;
```

Comment:
Can use ReactNode

Reply:
ReactNode supports a much wider range of types which are `string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | React.ReactFragment | React.ReactPortal | null | undefined`.
There's no need to add all these types while these 2 are enough.